### PR TITLE
Remove guava

### PR DESCRIPTION
**Pull request by CRogers**
## Before this PR
As per #147, the guava dependency is making it hard for external people to use this plugin.

## After this PR
==COMMIT_MSG==
Guava dependency is entirely removed to remove dependency conflicts.
==COMMIT_MSG==

## Possible downsides?
Code is less concise when we could have used guava instead.

### CODE REVIEW

**Comment by dansanduleac** [comment]
I'm not so sure we want this, as this is not a basic plugin and some of the logic now looks uglier.
Can we consider shading guava instead?

File: src/main/java/com/palantir/gradle/revapi/config/GroupNameVersion.java [hunk 1]
```diff
@@ -37,17 +36,17 @@ default String asString() {
 
     @JsonCreator
     static GroupNameVersion fromString(String groupNameVersionString) {
-        List<String> split = Splitter.on(':').splitToList(groupNameVersionString);
+        String[] split = groupNameVersionString.split(":");
```
Comment (dansanduleac) [inline src/main/java/com/palantir/gradle/revapi/config/GroupNameVersion.java:39]: this has a bug where `::` will be considered as a single separator, I don't really want to go back to a world where we have to think about these bugs

**Comment by dansanduleac** [comment]
@CRogers thoughts on just trying to shade guava, or perhaps offer a separate coordinate where we shade it, for people who are hitting these issues?

**Comment by CRogers** [comment]
Shading would be easier. The downside is shading guava adds about 4mb to the jar size, but this probably isn't that important. Ideally we'd open source that shading library we use internally to be able to do it.

File: src/main/java/com/palantir/gradle/revapi/config/GroupNameVersion.java (hunk 1 above)
Comment (iamdanfox) [reply src/main/java/com/palantir/gradle/revapi/config/GroupNameVersion.java:39]: Alternative here could be to just use a regex to require the (group):(name):(version) thing we expect?

**Comment by iamdanfox** [comment]
Given that a second person has had trouble with guava on their plugin classpath, can we just merge this?

**Review by iamdanfox — APPROVED** [review]

**Comment by CRogers** [comment]
I would be pro just merging it

**Comment by CRogers** [comment]
:+1:

**Comment by dansanduleac** [comment]
Do as you like, but I think if we merge this then the same argument applies to all of our other plugins.
Ultimately we do rely quite heavily on guava and will likely see more bugs by removing it.

**Comment by CRogers** [comment]
Tbh, I am pro not using guava where we can get away with it. Otherwise we need to prioritise getting `gradle-shadow-jar` open sourced so we can do some easy partial shading.

**Pull request merged**